game-core/src/test/java/games/strategy/triplea/delegate/battle/MustFightBattleExecutablesTest.java

```diff
@@ -0,0 +1,1670 @@
+  }
+
+  @Test
+  @DisplayName(
+      "Verify defending transports are not removed if "
+          + "TRANSPORT_CASUALTIES_RESTRICTED is true but no defenders")
+  void defendingTransportsAreNotRemovedIfTransportCasualtiesRestrictedButNoDefenders() {
+    final MustFightBattle battle = spy(newBattle(WATER));
+    doReturn(List.of()).when(battle).getAttackerRetreatTerritories();
+    when(gameProperties.get(SUB_RETREAT_BEFORE_BATTLE, false)).thenReturn(false);
+    when(gameProperties.get(TRANSPORT_CASUALTIES_RESTRICTED, false)).thenReturn(true);
+
+    final Tuple<Unit, UnitAttachment> unitAndAttachment = newUnitAndAttachment();
+    final Unit unit = unitAndAttachment.getFirst();
+    when(unit.getOwner()).thenReturn(defender);
+    final UnitAttachment attachment1 = unitAndAttachment.getSecond();
+    when(attachment1.getIsCombatTransport()).thenReturn(false);
+    when(attachment1.getTransportCapacity()).thenReturn(2);
+    when(attachment1.getIsSea()).thenReturn(true);
+
+    final Tuple<Unit, UnitAttachment> unitAndAttachment2 = newUnitAndAttachment();
+    final Unit unit2 = unitAndAttachment2.getFirst();
+    when(unit2.getOwner()).thenReturn(attacker);
+    final UnitAttachment attachment2 = unitAndAttachment2.getSecond();
+    when(attachment2.getMovement(defender)).thenReturn(0);
+    when(attachment2.getIsSea()).thenReturn(true);
+
+    when(battleSite.getUnits()).thenReturn(List.of(unit, unit2));
+
+    battle.setUnits(List.of(unit2), List.of(unit), List.of(), List.of(), defender, List.of());
+    final List<IExecutable> execs = battle.getBattleExecutables(true);
+
+    final int index = getIndex(execs, MustFightBattle.RemoveUndefendedTransports.class);
+    final IExecutable step = execs.get(index);
+
+    final IDelegateBridge delegateBridge = mock(IDelegateBridge.class);
+    step.execute(null, delegateBridge);
+
+    verify(delegateBridge, never()).addChange(any());
+    verify(battle, never()).remove(any(), any(), any(), eq(false));
+  }
+
+  @Test
+  @DisplayName(
+      "When attacker has a destroyer, defender has a destroyer, WW2v2 is true, "
+          + "and DEFENDING_SUBS_SNEAK_ATTACK is either, then attacker>defender>standard")
+  void firstStrikeOrderAttHasDestroyerDefHasDestroyerWW2v2TrueSneakAttackTrueFalse() {
+    assertThatFirstStrikeStepOrder(
+        givenFirstStrikeBattleSetup(true, true, true, true, true),
+        List.of(
+            FirstStrikeBattleStep.ATTACKER,
+            FirstStrikeBattleStep.DEFENDER,
+            FirstStrikeBattleStep.STANDARD));
+  }
+
+  private Tuple<MustFightBattle, List<IExecutable>> givenFirstStrikeBattleSetup(
+      final boolean attackerDestroyer,
+      final boolean defenderDestroyer,
+      final boolean ww2v2,
+      final boolean defendingSubsSneakAttack,
+      final boolean ignoreDefendingSubsSneakAttack) {
+    final MustFightBattle battle = spy(newBattle(WATER));
+    lenient().doNothing().when(battle).firstStrikeAttackersFire(any());
```

Comment:
Which cases invoke this lenient and which other ones do not? Is the lenient a sign that we could use multiple setup methods? I'm curious about this as lenient mocks can be a testing smell, perhaps an overly generic test scaffolding.

Reply:
The cases where `attacker == true` in `assertThatFirstStrikeReturnFireIs` will call the `firstStrikeAttackersFire` and when `attacker == false` it will call `firstStrikeDefendersFire`